specs-go/config.go

```diff
@@ -20,13 +20,32 @@ type Spec struct {
 	Hooks Hooks `json:"hooks"`
 	// Annotations is an unstructured key value map that may be set by external tools to store and retrieve arbitrary metadata.
 	Annotations map[string]string `json:"annotations,omitempty"`
-
+	// VM specifies configuration for virtual-machine-based runtimes.
+	VM VM `json:"vm,omitempty"`
 	// Linux is platform specific configuration for Linux based containers.
 	Linux Linux `json:"linux" platform:"linux,omitempty"`
 	// Solaris is platform specific configuration for Solaris containers.
 	Solaris Solaris `json:"solaris" platform:"solaris,omitempty"`
 }
 
+// VM contains information for virtual-machine-based runtimes.
+type VM struct {
+	// Kernel specifies kernel-related configuration for virtual-machine-based runtimes.
+	Kernel Kernel `json:"kernel"`
+	// ImagePath is the path to the root filesystem image on the host which can be used by a virtual-machine-based runtime.
+	ImagePath string `json:"imagePath"`
+}
+
+// Kernel contains information about the kernel to use for a virtual machine.
+type Kernel struct {
+	// Path is the path to the kernel used to boot the virtual machine.
+	Path string `json:"path"`
+	// Parameters specifies parameters to pass to the kernel.
+	Parameters string `json:"parameters,omitempty"`
+	// InitRd is the path to an initial ramdisk to be used by the kernel.
+	InitRd string `json:"initrd,omitempty"`
```

Comment:
Also, I forgot to mention that I'm still iffy about making the `initrd` a first-class citizen (but at least it's optional).